homeassistant/components/vallox/binary_sensor.py

```diff
@@ -0,0 +1,85 @@
+"""Support for Vallox ventilation unit binary sensors."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import ValloxDataUpdateCoordinator
+from .const import DOMAIN
+
+
+class ValloxBinarySensor(CoordinatorEntity, BinarySensorEntity):
+    """Representation of a Vallox binary sensor."""
+
+    entity_description: ValloxBinarySensorEntityDescription
+    coordinator: ValloxDataUpdateCoordinator
+
+    def __init__(
+        self,
+        name: str,
+        coordinator: ValloxDataUpdateCoordinator,
+        description: ValloxBinarySensorEntityDescription,
+    ) -> None:
+        """Initialize the Vallox binary sensor."""
+        super().__init__(coordinator)
+
+        self.entity_description = description
+
+        self._attr_name = f"{name} {description.name}"
+
+        uuid = self.coordinator.data.get_uuid()
+        self._attr_unique_id = f"{uuid}-{description.key}"
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return true if the binary sensor is on."""
+        if (metric_key := self.entity_description.metric_key) is None:
+            return None
+        return self.coordinator.data.get_metric(metric_key) == 1
+
+
+@dataclass
+class ValloxBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """Describes Vallox binary sensor entity."""
+
+    metric_key: str | None = None
```

Comment:
In sensor.py there is the case of the profile sensor which doesn’t need a metric key, because we have to query using get_profile().

As long as we don’t have a similar case for binary sensors, we could remove the None here, which simplifies the is_on() a bit.

Reply:
Mypy doesn't accept that: `error: Attributes without a default cannot follow attributes with one` (this comes from the attributes defined in the superclass EntityDescription)

Reply:
I think this should do the trick:

https://github.com/home-assistant/core/blob/1bd1a12285d92d46de2600fda1b8fce871423851/homeassistant/components/zwave_js/binary_sensor.py#L61-L72

Reply:
I'm not sure what you mean. The implementation in zwave_js seems quite a bit more complex as it allows multiple different values that mean "on" depending on sensor. As far as I can tell the Vallox API always uses 0 for off and 1 for on, so we shouldn't need that kind of flexibility here. Other than that the is_on method implementation there is exactly the same as the one we have here.

Reply:
I was referring to the double inheritance using the “Mixin”, which should allow you to get rid of the default None as originally mentioned.